sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide-WindowsAzure-ServiceBus.md

```diff
@@ -0,0 +1,473 @@
+    await args.CompleteMessageAsync(args.Message);
+
+    // we can also set arbitrary session state using this receiver
+    // the state is specific to the session, and not any particular message
+    await args.SetSessionStateAsync(new BinaryData("some state"));
+}
+
+Task ErrorHandler(ProcessErrorEventArgs args)
+{
+    // the error source tells me at what point in the processing an error occurred
+    Console.WriteLine(args.ErrorSource);
+    // the fully qualified namespace is available
+    Console.WriteLine(args.FullyQualifiedNamespace);
+    // as well as the entity path
+    Console.WriteLine(args.EntityPath);
+    Console.WriteLine(args.Exception.ToString());
+    return Task.CompletedTask;
+}
+
+// start processing
+await processor.StartProcessingAsync();
+
+// since the processing happens in the background, we add a Conole.ReadKey to allow the processing to continue until a key is pressed.
+Console.ReadKey();
+```
+
+The below code snippet shows you the session variation of the receiver. Please note that creating a session receiver is an async operation because the library will need to get a lock on the session by connecting to the service first.
+
+Create a receiver that will receive from the next available session:
+
+```C# Snippet:ServiceBusReceiveNextSession
+ServiceBusSessionReceiver receiver = await client.AcceptNextSessionAsync(queueName);
+
+// the received message is a different type as it contains some service set properties
+ServiceBusReceivedMessage receivedMessage = await receiver.ReceiveMessageAsync();
+Console.WriteLine(receivedMessage.SessionId);
+```
+
+Create a receiver that will receive from a specific session:
+
+```C# Snippet:ServiceBusReceiveFromSpecificSession
+// create a receiver specifying a particular session
+ServiceBusSessionReceiver receiver = await client.AcceptSessionAsync(queueName, "Session2");
+
+// the received message is a different type as it contains some service set properties
+ServiceBusReceivedMessage receivedMessage = await receiver.ReceiveMessageAsync();
+Console.WriteLine(receivedMessage.SessionId);
+```
+
+### Cross-Entity transactions
+
+Previously, in `WindowsAzure.ServiceBus`, when performing a transaction that spanned multiple queues, topics, or subscriptions you would need to use the "ViaPartitionKey" option
+in the `MessageSender`.
+
+Now in `Azure.Messaging.ServiceBus`, there is an `EnableCrossEntityTransactions` property on the `ServiceBusClientOptions`. When setting this property to `true`, the first operation that occurs using any senders or receivers created from the client implicitly becomes the send-via entity. Because of this, subsequent operations must either be by senders, or if they are by receivers, the receiver must be receiving from the send-via entity. For this reason, it probably makes more sense to have your first operation be a receive rather than a send when setting this property.

```

Comment:
```suggestion
In `Azure.Messaging.ServiceBus`, the `EnableCrossEntityTransactions` property on `ServiceBusClientOptions` serves this purpose. When setting this property to `true`, the first operation that occurs using any senders or receivers created from the client implicitly becomes the send-via entity. Because of this, subsequent operations must either be by senders, or if they are by receivers, the receiver must be receiving from the send-via entity. For this reason, it probably makes more sense to have your first operation be a receive rather than a send when setting this property.
```

Is there a doc that we can cross-link for the "send via" concept?   This confuses the heck out of people (myself included.)